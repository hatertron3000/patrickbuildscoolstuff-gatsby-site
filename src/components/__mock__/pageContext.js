module.exports = {
    "content": {
      "raw": "{\"nodeType\":\"document\",\"data\":{},\"content\":[{\"nodeType\":\"heading-2\",\"content\":[{\"nodeType\":\"text\",\"value\":\"Let me tell you about my tacos.\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"This is where you would \",\"marks\":[],\"data\":{}},{\"nodeType\":\"text\",\"value\":\"write \",\"marks\":[{\"type\":\"italic\"}],\"data\":{}},{\"nodeType\":\"text\",\"value\":\"about \",\"marks\":[{\"type\":\"underline\"}],\"data\":{}},{\"nodeType\":\"text\",\"value\":\"tacos\",\"marks\":[{\"type\":\"bold\"}],\"data\":{}},{\"nodeType\":\"text\",\"value\":\". They are really great and stuff.\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"embedded-asset-block\",\"content\":[],\"data\":{\"target\":{\"sys\":{\"id\":\"5Xz9NRMUycoH12tz3sWWHK\",\"type\":\"Link\",\"linkType\":\"Asset\"}}}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"But those are pictures of hot dogs. Oh no!\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"heading-3\",\"content\":[{\"nodeType\":\"text\",\"value\":\"What will we ever do?\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"embedded-asset-block\",\"content\":[],\"data\":{\"target\":{\"sys\":{\"id\":\"3EZ9oeKZ9trczs6RqJvTui\",\"type\":\"Link\",\"linkType\":\"Asset\"}}}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"Ah, that's better.\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"unordered-list\",\"content\":[{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"Here\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"are\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"some\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"bullets\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"ordered-list\",\"content\":[{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"and\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"numbers\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"list-item\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"too\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"blockquote\",\"content\":[{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"I am a blockquote\",\"marks\":[],\"data\":{}}],\"data\":{}}],\"data\":{}},{\"nodeType\":\"hr\",\"content\":[],\"data\":{}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"I'm below an HR\",\"marks\":[],\"data\":{}}],\"data\":{}},{\"nodeType\":\"paragraph\",\"content\":[{\"nodeType\":\"text\",\"value\":\"\",\"marks\":[],\"data\":{}}],\"data\":{}}]}",
      "references": [
        {
          "id": "b0901419-16a3-5001-adc1-4ebd43e137cb",
          "contentful_id": "5Xz9NRMUycoH12tz3sWWHK",
          "gatsbyImageData": {
            "images": {
              "sources": [
                {
                  "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=1680&h=1120&q=50&fm=webp 1680w,\n//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=3360&h=2240&q=50&fm=webp 3360w,\n//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=4000&h=2666&q=50&fm=webp 4000w",
                  "sizes": "(min-width: 6720px) 6720px, 100vw",
                  "type": "image/webp"
                }
              ],
              "fallback": {
                "src": "//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=1680&h=1120&fl=progressive&q=50&fm=jpg",
                "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=1680&h=1120&fl=progressive&q=50&fm=jpg 1680w,\n//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=3360&h=2240&fl=progressive&q=50&fm=jpg 3360w,\n//images.ctfassets.net/fp1mn1nwu0hx/5Xz9NRMUycoH12tz3sWWHK/40cf55edde6f74f80afcc5db604f3fb1/ball-park-brand-RKQ4-Q5FF-o-unsplash.jpg?w=4000&h=2666&fl=progressive&q=50&fm=jpg 4000w",
                "sizes": "(min-width: 6720px) 6720px, 100vw"
              }
            },
            "layout": "constrained",
            "width": 6720,
            "height": 4480
          }
        },
        {
          "id": "e1b50f48-b22a-5b55-a05f-51cac02eed21",
          "contentful_id": "3EZ9oeKZ9trczs6RqJvTui",
          "gatsbyImageData": {
            "images": {
              "sources": [
                {
                  "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=56&h=56&q=50&fm=webp 56w,\n//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=113&h=113&q=50&fm=webp 113w,\n//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=225&h=225&q=50&fm=webp 225w",
                  "sizes": "(min-width: 225px) 225px, 100vw",
                  "type": "image/webp"
                }
              ],
              "fallback": {
                "src": "//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=225&h=225&fl=progressive&q=50&fm=jpg",
                "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=56&h=56&fl=progressive&q=50&fm=jpg 56w,\n//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=113&h=113&fl=progressive&q=50&fm=jpg 113w,\n//images.ctfassets.net/fp1mn1nwu0hx/3EZ9oeKZ9trczs6RqJvTui/d9c1ff025228aaa3f450d0b642d750f7/download.jpg?w=225&h=225&fl=progressive&q=50&fm=jpg 225w",
                "sizes": "(min-width: 225px) 225px, 100vw"
              }
            },
            "layout": "constrained",
            "width": 225,
            "height": 225
          }
        }
      ]
    },
    "headline": "Tacos",
    "summary": {
      "summary": "Tacos really are the best."
    },
    "publishedDate": new Date(1617753600000),
    "slug": "tacos",
    "thumbnail": {
      "gatsbyImageData": {
        "images": {
          "sources": [
            {
              "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=1289&h=1289&q=50&fm=webp 1289w,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=2578&h=2578&q=50&fm=webp 2578w,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=4000&h=4000&q=50&fm=webp 4000w",
              "sizes": "(min-width: 5156px) 5156px, 100vw",
              "type": "image/webp"
            }
          ],
          "fallback": {
            "src": "//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=1289&h=1289&fl=progressive&q=50&fm=jpg",
            "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=1289&h=1289&fl=progressive&q=50&fm=jpg 1289w,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=2578&h=2578&fl=progressive&q=50&fm=jpg 2578w,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=4000&h=4000&fl=progressive&q=50&fm=jpg 4000w",
            "sizes": "(min-width: 5156px) 5156px, 100vw"
          }
        },
        "layout": "constrained",
        "width": 5156,
        "height": 5156
      },
      "title": "Tacos",
      "fixed": {
        "src": "//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=250&q=50",
        "srcSet": "//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=250&h=250&q=50 1x,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=375&h=375&q=50 1.5x,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=500&h=500&q=50 2x,\n//images.ctfassets.net/fp1mn1nwu0hx/jj5h83LUtDnkFpKikl4F4/659142478200c269855f514ab9b01419/jarritos-mexican-soda-20y_7PWsgmk-unsplash.jpg?w=750&h=750&q=50 3x",
        "width": 250,
        "height": 250
      }
    }
  }