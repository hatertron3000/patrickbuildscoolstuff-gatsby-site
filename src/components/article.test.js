import React from "react"
import renderer from "react-test-renderer"
import {documentToReactComponents} from '@contentful/rich-text-react-renderer'
import pageContext  from './__mock__/pageContext'

import Article from "./article"

describe("Article", () => {
    it("matches snapshot", () => {
        const {
            content,
            headline,
            publishedDate,
            thumbnail
        } = pageContext

        const tree = renderer
            .create(<Article 
                content={documentToReactComponents(JSON.parse(content.raw))}
                headline={headline}
                publishedDate={publishedDate}
                thumbnail={thumbnail}
            />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})