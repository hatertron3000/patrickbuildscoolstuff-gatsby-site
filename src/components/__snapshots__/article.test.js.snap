// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article matches snapshot 1`] = `
<div>
  <h1>
    Tacos
  </h1>
  <div>
    <p>
      Sun Apr 04 2021 20:51:00 GMT-0500 (Central Daylight Time)
    </p>
  </div>
  <div>
    <h2>
      Let me tell you about my tacos.
    </h2>
    <p>
      This is where you would 
      <i>
        write 
      </i>
      <u>
        about 
      </u>
      <b>
        tacos
      </b>
      . They are really great and stuff.
    </p>
    <p>
      But those are pictures of hot dogs. Oh no!
    </p>
    <h3>
      What will we ever do?
    </h3>
    <p>
      Ah, that's better.
    </p>
    <ul>
      <li>
        <p>
          Here
        </p>
      </li>
      <li>
        <p>
          are
        </p>
      </li>
      <li>
        <p>
          some
        </p>
      </li>
      <li>
        <p>
          bullets
        </p>
      </li>
    </ul>
    <p>
      
    </p>
    <ol>
      <li>
        <p>
          and
        </p>
      </li>
      <li>
        <p>
          numbers
        </p>
      </li>
      <li>
        <p>
          too
        </p>
      </li>
    </ol>
    <p>
      
    </p>
    <blockquote>
      <p>
        I am a blockquote
      </p>
    </blockquote>
    <hr />
    <p>
      I'm below an HR
    </p>
    <p>
      
    </p>
  </div>
</div>
`;
